contracts/LyraL2/LyraController.sol

```diff
@@ -0,0 +1,215 @@
+            );
+        } else {
+            optionType = isCall
+                ? LyraAdapter.OptionType.LONG_CALL
+                : LyraAdapter.OptionType.LONG_PUT;
+
+            /// Use params to open new position on lyra
+            openPositionParams = _getTradeInputParams(
+                strikeId,
+                0,
+                optionType,
+                amount,
+                sUSD.balanceOf(address(this))
+            );
+        }
+
+        tradeResult = LyraAdapter._openPosition(openPositionParams);
+
+        uint256 sUSDSent = sUSDBal - sUSD.balanceOf(address(this));
+
+        currentPosition = CurrentPosition({
+            strikeId: strikeId,
+            positionId: tradeResult.positionId,
+            optionType: optionType,
+            amount: sUSDSent,
+            optionsPurchased: amount,
+            isActive: true
+        });
+    }
+
+    /// @notice Exercises/Sell option on lyra.
+    /// @dev Will sell back or settle the option on Lyra.
+    /// @param toSettle boolean if true settle position, else close position
+    function _closePosition(bool toSettle) internal {
+        require(currentPosition.isActive, "NO_ACTIVE_POSITION");
+
+        /// Check if option has to be settled.
+        if (toSettle == true) {
+            uint256[] memory positionsToClose = new uint256[](1);
+            positionsToClose[0] = currentPosition.positionId;
+
+            LyraAdapter.shortCollateral.settleOptions(positionsToClose);
+        } else {
+            LyraAdapter.TradeInputParameters
+                memory closePositionParams = _getTradeInputParams(
+                    currentPosition.strikeId,
+                    currentPosition.positionId,
+                    currentPosition.optionType,
+                    currentPosition.optionsPurchased,
+                    type(uint256).max
+                );
+            LyraAdapter._closeOrForceClosePosition(closePositionParams);
+        }
+
+        // reset the current position value.
+        currentPosition = CurrentPosition({
+            strikeId: 0,
+            positionId: 0,
+            optionType: LyraAdapter.OptionType.SHORT_CALL_BASE,
+            amount: 0,
+            optionsPurchased: 0,
+            isActive: false
+        });
+    }
+
+    /// @notice Get the value of current active position on Lyra.
+    /// @dev Gives the total value of position handler in susd.
+    function _positionInWantToken() public view returns (uint256) {
+        uint256[] memory positions = new uint256[](1);
+        positions[0] = currentPosition.positionId;
+
+        LyraAdapter.OptionPosition[] memory allPositions = LyraAdapter
+            ._getPositions(positions);
+        LyraAdapter.OptionPosition memory positionData = allPositions[0];
+
+        if (positionData.state == LyraAdapter.PositionState.ACTIVE) {
+            (uint256 callPremium, uint256 putPremium) = LyraAdapter
+                ._getPurePremiumForStrike(currentPosition.strikeId);
```

Comment:
@Bapi-Reddy this wont work for getting option price. What do we do in this case? For position in want token